examples/benchmark/thrift/client.py

```diff
@@ -50,26 +52,28 @@ def report_work():
 
 @gen.coroutine
 def do_work():
-    global requests
-
-    # TODO: make this configurable
-    data = os.urandom(4096)
-
     while True:
         yield tchannel.thrift(
             request=kv.setValue("key", data),
         )
-        local.requests += 1
 
-        # TODO: get/set ratio
-        yield tchannel.thrift(
-            request=kv.getValue("key"),
-        )
         local.requests += 1
 
 
 if __name__ == '__main__':
+    if len(sys.argv) > 1:
+        concurrency = int(sys.argv[1])
+    else:
+        concurrency = 100
+
+    sys.stderr.write('using concurrency %s\n' % concurrency)
+    sys.stderr.flush()
+
+    for _ in xrange(concurrency):
+        do_work()
+
     ioloop.PeriodicCallback(report_work, 1000).start()
+
     try:
         ioloop.IOLoop.current().run_sync(do_work)
```

Comment:
You want to block until all the `do_work`s are done, right? You need to use `tornado.gen.multi_future` on the list of futures to get back a future that finishes when all the futures from the list are done. So, something like,

    futures = [do_work() for _ in xrange(concurrency)]
    future = multi_future(futures)

    io_loop.add_future(future, lambda _: io_loop.stop())
    io_loop.start()